common/djangoapps/course_modes/models.py

```diff
@@ -666,6 +672,13 @@ def __unicode__(self):
         )
 
 
+@receiver(models.signals.post_save, sender=CourseMode)
+@receiver(models.signals.post_delete, sender=CourseMode)
+def invalidate_course_mode_cache(sender, **kwargs):   # pylint: disable=unused-argument
+    """Invalidate the cache of course modes. """
+    RequestCache.clear_request_cache(name=CourseMode.CACHE_NAMESPACE)
```

Comment:
Do we need to clear out the entire cache here, or would it be better to check `kwargs['instance']` to clear out only the course in question?

[docs](https://docs.djangoproject.com/en/1.8/ref/signals/#post-save) indicate `instance` should be available for either of these signals.